### ISSUE TITLE
Parent / Child / Sibling post sytstem

### ISSUE THREAD

**Issue by ghost** [Onani-Dev/Onani#121]
This also serves as implementation detail, so post your suggestions here

**Comment by kapsikkum** [comment]
THIS NEEDS TO BE DONE!!!!!!!!!!!!!!!

**Comment by ghost** [comment]
ok but how will it work???

**Comment by kapsikkum** [comment]
parent gets children
parent post -> children (can also be their own parents of other posts)

**Comment by ghost** [comment]
what if we just want some posts to be siblings without a parent??!

**Comment by kapsikkum** [comment]
did not think of this.

**Comment by ghost** [comment]
Can posts have multiple parents???

**Comment by kapsikkum** [comment]
i don't think that would be wise

**Comment by ghost** [comment]
It's a huge mess lol, probably toughest thing to design right out of all of onani, hence why i did not rush it

**Comment by kapsikkum** [comment]
what if we had only siblings?

**Comment by ghost** [comment]
Would be a mess database-wise init
Also if post A is a sibling of post B, and post B a sibling of post C, is post A a sibling of post C automatically?

**Comment by kapsikkum** [comment]
that would be wise yes

**Comment by ghost** [comment]
Would work pretty poorly with manips / redraws / edits, as they shouldn't be along the og siblings

**Comment by kapsikkum** [comment]
1 parent, unlimited (limited maybe idk) siblings

**Comment by ghost** [comment]
siblings are put there by hand?

**Comment by kapsikkum** [comment]
siblings are put there by multiple file uploads, and yeah by hand if needed

**Comment by ghost** [comment]
will become hellish mess!

**Comment by kapsikkum** [comment]
alterations with minimal changes should only have 1 post visible so i'm not sure how that would work.

**Comment by kapsikkum** [comment]
what are your thoughts on how it should be implemented?

**Comment by ghost** [comment]
> alterations with minimal changes should only have 1 post visible so i'm not sure how that would work.

the MyImouto way, separate "shown in index" field



> what are your thoughts on how it should be implemented?

I actually have no idea lol
The main use case for parent posts is alts, but it's not so obvious which post should be the parent
Imo it should be closer to pools, but that would become a cumbersome to use mess

**Comment by kapsikkum** [comment]
smaller groups of posts?

**Comment by ghost** [comment]
What do you mean?

**Comment by kapsikkum** [comment]
> What do you mean?

have pool/collection like objects in the database that have posts in a one to many relationship, so that the posts in that group can see the siblings and the group could have an index on what the main post is etc

**Comment by ghost** [comment]
so literally just danbooru's parent/child system?

**Comment by kapsikkum** [comment]
More like mini collections but they have no names and their purpose is to keep track of posts and their relationships with other posts

**Comment by ghost** [comment]
yeah that was my idea, but that's exactly what i meant by "it would get cumbersome"
How do you reference these? Do they have an id like pools? Since they're just pools without the name, why not use that instead?

**Comment by kapsikkum** [comment]
It wouldn't be seen by the users, only the data that it holds. Say a post had a parent and 3 siblings, the data of that group object would say the parent is post 1 and the children/siblings are the rest

**Comment by ghost** [comment]
I think we should start from scratch and look at the use cases for the feature instead of suggesting stuff

**Comment by kapsikkum** [comment]
I will do a test implementation of my idea when I can and put it in another branch

**Comment by ghost** [comment]
I'd say just having a regular Parent/Post system (with siblings automatically found by their common parent) would probably work well enough, and would be the easiest one to migrate **from** in the future
